### ISSUE TITLE
HttpClient specific ssl configuration is not recognized

### ISSUE THREAD

**Issue by chrisingenhaag** [micronaut-projects/micronaut-core#2231]
Hello,
we have an application with two declarative http clients configured via application.yaml. One of this services requires a ssl client certificate authentication. Our target is to configure ssl just for this services and not for the other one. But key and truststore config is not loaded at all. We debugged a bit through micronaut httpclient creation an configuration options don´t seem to be recognized.

But if we configure the default client configuration, it works. But then the stores would be used for both services.

### Task List

- [x] Steps to reproduce provided
- [ ] Stacktrace (if present) provided
- [ ] Example that reproduces the problem uploaded to Github
- [x] Full description of the issue provided (see below)

### Steps to Reproduce

application.yml which is **not** working
```
micronaut:
  # ...
  http:
    services:
      first-service:
        url: `https://first-service.com`
        ssl:
          enabled: false
      second-service:
        url: `https://second-service.com`
        path: /api
        ssl:
          enabled: true
          client-authentication: NEED
          trust-store:
            path: file:/path/to/truststore.jks
            password: foobar
            type: JKS
          key:
            alias: 1
            password: foobar
          key-store:
            path: file:/path/to/keystore.jks
            password: foobar
            type: JKS
```

application.yml which is working, but uses default config for all http clients:
```
micronaut:
  # ...
  http:
    client:
      ssl:
        enabled: true
        client-authentication: WANT
        trust-store:
          path: file:/path/to/truststore.jks
          password: foobar
          type: JKS
        key:
          alias: 1
          password: foobar
        key-store:
          path: file:/path/to/keystore.jks
          password: foobar
          type: JKS
    services:
      first-service:
        url: `https://first-service.com`
        ssl:
          enabled: false
      second-service:
        url: `https://second-service.com`
        path: /api
        ssl:
          enabled: true
          client-authentication: NEED
```

### Expected Behaviour

Truststore and Keystore are loaded when configured for a specific service and used as sslcontext for https client requests where client certificate authentication is required.

### Actual Behaviour

No key- and truststore is loaded and because of this client certificate authentication is not working.

### Environment Information

- **Operating System**: docker openjdk:11-jre-slim
- **Micronaut Version:** 1.2.2
- **JDK Version:** 11

**Comment by jameskleeh** [comment]
Duplicate of https://github.com/micronaut-projects/micronaut-core/issues/2229

**Comment by chrisingenhaag** [comment]
@jameskleeh  Is this really a duplicate? In our case no error is thrown. Applications works correctly but only ssl stores are not available.

**Comment by jameskleeh** [comment]
@chrisingenhaag As far as I can tell yes I believe its a duplicate. You can test the snapshot the build after the other issues fix has been pushed to verify

**Comment by olevold** [comment]
How is this fixed? I'm running Micronaut 2.5.11 and I'm experiencing exactly the same problem as chrisingenhaag
